### PR TITLE
Add WiX support to OneLocBuild

### PR DESCRIPTION
**Pull request by jonfortescue**
Makes the generate-locprojectjson script pick up wxl files for WiX localization. Example of the generated locproject.json [here](https://github.com/jonfortescue/AspNetCore/blob/WxlLocTest/eng/Localize/LocProject.json). Sample PR [here](https://github.com/dotnet/aspnetcore/pull/43203).

Closes https://github.com/dotnet/arcade/issues/10298.

### CODE REVIEW

File: eng/common/generate-locproject.ps1 [hunk 1]
```diff
@@ -88,6 +89,32 @@ $locJson = @{
                     }
                 }
             )
+        },
+        @{
+            LanguageSet = "cs-CZ;de-DE;es-ES;fr-FR;it-IT;ja-JP;ko-KR;pl-PL;pt-BR;ru-RU;tr-TR;zh-CN;zh-TW"
```
Comment (dougbu) [inline eng/common/generate-locproject.ps1:94]: Pity https://ceapex.visualstudio.com/CEINTL/_git/SoftwareLocalization?path=/src/OneBranchPackages/Localization.Languages/Localization.Languages.props lacks a language set matching this.

**Review by dougbu — DISMISSED** [review]
Sample project looks good 🎉

**Comment by jonfortescue** [comment]
@dougbu here's a [sample PR](https://github.com/dotnet/aspnetcore/pull/43203) it opened to aspnetcore.

File: eng/common/generate-locproject.ps1 [hunk 2]
```diff
@@ -33,6 +33,8 @@ $jsonTemplateFiles | ForEach-Object {
 
 $jsonWinformsTemplateFiles = Get-ChildItem -Recurse -Path "$SourcesDirectory" | Where-Object { $_.FullName -Match "en\\strings\.json" } # current winforms pattern
 
+$wxlFiles = Get-ChildItem -Recurse -Path "$SourcesDirectory" | Where-Object { $_.FullName -Match "\\.+\.wxl" -And -Not( $_.Directory.Name -Match "\d{4}" ) }
```
Comment (mmitche) [inline eng/common/generate-locproject.ps1:36]: Can you put a comment here as to why the \d{4} exclusion is here?

Comment (jonfortescue) [reply eng/common/generate-locproject.ps1:36]: yup yup

Comment (jonfortescue) [reply eng/common/generate-locproject.ps1]: done

**Comment by dougbu** [comment]
> @dougbu here's a [sample PR](https://github.com/dotnet/aspnetcore/pull/43203) it opened to aspnetcore.

Looks like we need to add exclusions to avoid e.g. https://github.com/dotnet/aspnetcore/blob/afd459e52b8239fa819bf52bcd35c3da72042eeb/src/Installers/Windows/AspNetCoreModule-Setup/setupstrings.wxl. According to a @joeloff, we only want / need to localize the hosting _bundle_ installers. Could you remind me of the exclusion syntax❔

----

Separately, what's the best way to inform the loc team that the repo already contains translations most of the strings (the case in all repos I looked at)❔ /cc @vlada-shubina @cristianosuzuki77

**Comment by joeloff** [comment]
@dougbu did I really say we only care about the hosting bundle? ASP.NET is building separate bundles for just the runtime as I recall, and we have the SDK/desktop/runtime bundles too

**Comment by dougbu** [comment]
Sorry @joeloff, I should have emphasized the word "bundle" rather than "hosting". That's the gist of what you said and our hosting bundle was an example.

@jonfortescue the overall point here is we have WXL files that aren't worth localization.

**Review by garath — APPROVED** [review]

**Comment by dougbu** [comment]
Curious: Will the `Test XHarness SDK Apple_Simulators Build_Debug` job likely pass on retry❔ Or does the **Right Thing:tm:** require #10380 i.e. a full validation run❔

**Comment by MattGal** [comment]
> Curious: Will the `Test XHarness SDK Apple_Simulators Build_Debug` job likely pass on retry❔ Or does the **Right Thing™️** require #10380 i.e. a full validation run❔

Xharness flakiness is unfortunately just a sign of flaky on-prem devices, unless you're touching the XHarness SDK you should comfortably ignore failures and retry, letting us know if it's persistently failing (because this usually means a machine gone bad)

**Comment by jonfortescue** [comment]
Given everything here is passing and seems to be working, I'm merging.

**Pull request merged**